### PR TITLE
Channel fixes

### PR DESCRIPTION
**Pull request by or-else**
* `{get sub}` did not return user's own subscription in case of channel readers (no subscriptions were returned).
* `{note what=read}` did not get stored for channel readers, so the channel always had unread messages.
* There was inconsistent checking for channel vs group topic access.

### CODE REVIEW

**Comment by or-else** [comment]
A related note: the fact that channel subscriptions are not added to `topic.perUser` seems to create some complexity. I'm thinking about maybe actually adding them bun only while there are active sessions. What do you think?

**Comment by or-else** [comment]
@aforge Ping

File: server/topic.go [hunk 1]
```diff
@@ -930,60 +947,83 @@ func (t *Topic) handleBroadcast(msg *ServerComMessage) {
...
+				}
 
+				upd := map[string]interface{}{}
+				if recv > 0 {
+					upd["RecvSeqId"] = recv
+				}
+				if read > 0 {
+					upd["ReadSeqId"] = read
+				}
+				if err := store.Subs.Update(topicName, asUid, upd, false); err != nil {
```
Comment (aforge) [inline server/topic.go:1001]: (Perhaps that won't happen often if at all) but do we need to update subs here if `upd` is empty?

**Review by aforge — APPROVED** [review]

File: server/topic.go (hunk 1 above)
Comment (or-else) [reply server/topic.go:1001]: It cannot be empty: `seq > 0` means wither read or recv or both are > 0.

**Pull request merged**